linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/README.md

```diff
@@ -229,12 +228,14 @@ we get
 ```wat
 (type $v.A (sub $v.java.lang.Object (struct
   ;; ... class metadata
+  ;; ... itable slots
```

Comment:
Just to check my understanding: The itable slots *must* be before the method dispatch references for the vtable subtyping relaionship to hold.

Reply:
Yes, that's right. It's stronger than the Wasm subtyping, though. Even if we were targeting native code, where subtyping is not a thing, we would have to it this way. All classes have the same number of itable slots, but they have a different number of vtable method pointers. I we put method pointers first, either the itable slots would not be at a constant offset, or we would have to pad all method pointer tables to have the same (maximum) length.